### PR TITLE
Bug 1986981: Alert Config update - Patch to PR#1310

### PR DESCRIPTION
**Pull request by raptorsun**
This PR includes 2 amendments to PR#1310:
1. This PR patches Thanos-related alerts to warning severity and triggering time window to 1h. In PR#1310 I have introduced a bug in the patching function that does not take empty alert/record name as an all-matching wildcard. 😅
2. remove patch for "expr" field in rules KubePersistentVolumeFillingUp, NodeFilesystemFilesFillingUp, and NodeFilesystemSpaceFillingUp, leaving it to upstream

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by paulfantom** [comment]
`jsonnet/patch-rules.libsonnet` was moved to https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/utils/sanitize-rules.libsonnet

**Comment by dgrisonnet** [comment]
Holding until comments in the precedent PR are addressed: https://github.com/openshift/cluster-monitoring-operator/pull/1310#pullrequestreview-727253511

**Comment by raptorsun** [comment]
> `jsonnet/patch-rules.libsonnet` was moved to https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/utils/sanitize-rules.libsonnet

Thanks, now aligned to new master head.

**Comment by raptorsun** [comment]
Need to merge [this PR](https://github.com/openshift/cluster-monitoring-operator/pull/1299) to fix the test ci/prow/versions

**Comment by raptorsun** [comment]
The last commit about distinguishing rules under same name using index is moved to [another PR](https://github.com/openshift/cluster-monitoring-operator/pull/1331).
This PR now contains only changes in alert configuration.

File: jsonnet/utils/sanitize-rules.libsonnet [hunk 1]
```diff
@@ -342,7 +256,8 @@ local patchOrExcludeRule(rule, ruleSet, operation) =
   if std.length(ruleSet) == 0 then
     [rule]
   else if ('alert' in rule) then
-    local matchedRules = std.filter(function(ruleItem) ('alert' in ruleItem) && (ruleItem.alert == rule.alert), ruleSet);
+    // empty alert name is matching-all
```
Comment (dgrisonnet) [inline jsonnet/utils/sanitize-rules.libsonnet:259]: we shouldn't need to update the `patchOrExcludeRule` function if we only stick to updating the severities so would you mind reverted the changes you made to it in https://github.com/openshift/cluster-monitoring-operator/pull/1310?

Comment (raptorsun) [reply jsonnet/utils/sanitize-rules.libsonnet:259]: of course, this will be restored to previous version

Comment (raptorsun) [reply jsonnet/utils/sanitize-rules.libsonnet]: restored now :)

**Review by dgrisonnet — APPROVED** [review]
Awesome, looks good from my side, thank you @raptorsun for keeping up with my requests.

/lgtm
/unhold

**Pull request merged**